cpp/include/graph.hpp

```diff
@@ -661,4 +663,179 @@ template <typename ET>
 struct invalid_edge_id : invalid_idx<ET> {
 };
 
+namespace mg {
+/**
+ * @brief              A Replicatable Graph representation to replicate
+ *                     a graph using the handle
+ *
+ * @tparam VT              Type of vertex id
+ * @tparam ET              Type of edge id
+ * @tparam WT              Type of weight
+ * @tparam GraphTypeView   Type of Graph View
+ */
+template <typename VT, typename ET, typename WT, typename GraphTypeView>
+class ReplicatableGraph {
```

Comment:
Would it be possible/easier to have the replication done on the python side? That way we won't need to add another class. Is there anything that the c++ layer makes it better to do this as opposed to the python layer?

Reply:
> Are the current CI failures related to this?

Yes, the current CI failures should be fixed after merging #1008.

Reply:
Ideally, we should be able to efficiently replicate DataFrames using DASK. Not sure about the reality (and if I am not mistaken, you complained about performance issue using DASK), but if the performance difference is not dramatic, relying on DASK leads to more readable and maintainable code.

Reply:
The current version leverage the code currently present for MG - Distributed at the Python level, it would require some further refining that could be operated in future work.

Reply:
Un-resolving.

I agree with Aatish and Seunghwa on avoiding adding these C++ graph classes in graph.hpp. 
We were able to reuse the existing Di/Graph classes for MG distributed graphs, we should find a way for Batch as well. 

For replication in C++, the motivation was performance because the cost of the initial version in python was prohibitive. Let's continue the discussion on slack to find the best tradeoff between performance, UX, and specific code.

Reply:
The replication is now done at the Python side, the C++ class has been removed.